### PR TITLE
Balder/cross domain dependency

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/transactionlog/domain.cpp [hunk 1]
```diff
@@ -354,16 +354,17 @@ int Domain::closeSession(int sessionId)
...
             sessionRunTime = (std::chrono::steady_clock::now() - found->second->getStartTime());
             retval = 1;
-            _sessionExecutor.sync();
         }
     }
-    if (retval == 1) {
+    while (retval == 1) {
         FastOS_Thread::Sleep(10);
-        LockGuard guard(_sessionLock);
+        MonitorGuard guard(_sessionLock);
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domain.cpp]: Why change from LockGuard to MonitorGuard ?

File: searchlib/src/vespa/searchlib/transactionlog/domain.cpp [hunk 2]
```diff
@@ -354,16 +354,17 @@ int Domain::closeSession(int sessionId)
...
         SessionList::iterator found = _sessions.find(sessionId);
         if (found != _sessions.end()) {
-            _sessions.erase(sessionId);
-            retval = 0;
+            if ( ! found->second->isVisitRunning()) {
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/domain.cpp]: This does not appear to be safe.

File: searchlib/src/vespa/searchlib/transactionlog/domain.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domain.cpp]: Unintentional - reverted

File: searchlib/src/vespa/searchlib/transactionlog/domain.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/domain.cpp]: Lifetimes are handled by shared pointers, so I think we are good to go.

**Pull request merged**